index.user.js

```diff
@@ -6,42 +6,132 @@
+        `[style*="/profile_images/"] { display: none }`
+      ]
+    },
+    obfuscateName: {
+      default: true,
+      styles: [
+        `[data-testid="tweet"] [href^="/"]:not([aria-hidden]):not([href*="/status/"]),
+         [data-testid="UserCell"] [href^="/"]:not([href*="/status/"]) {
+            filter: blur(3px)
+         }
+        `
+      ]
+    },
+    hideName: {
+      default: false,
+      styles: [
+        `[data-testid="tweet"] [href^="/"]:not([aria-hidden]):not([href*="/status/"]),
+         [data-testid="UserCell"] [href^="/"]:not([href*="/status/"]) {
+            display: none
+         }
+        `
+      ]
     }
   }
 
-  function setFeatures(features) {
-    document.documentElement.setAttribute('data-refined-twitter-lite', features.join(' '))
-  }
+  const storageKey = 'refined-twitter-lite'
+  let settings = {}
+  const storedSettings = JSON.parse(localStorage.getItem(storageKey)) || {}
+  settings = Object.keys(features).reduce((settings, feature) => {
+    if (typeof storedSettings[feature] === 'boolean') {
+      settings[feature] = storedSettings[feature]
+    } else {
+      settings[feature] = features[feature].default
+    }
+    return settings
+  }, {})
+  console.log(storedSettings)
 
-  function onNavigate(url) {
+  function setFeatures() {
     const parsedUrl = document.createElement('a')
-    parsedUrl.href = url
-    setFeatures(
-      Object.keys(features).filter(feature => features[feature](parsedUrl, document.title || ''))
+    parsedUrl.href = window.location.href
+
+    const enabledFeatures = Object.keys(features).filter(feature =>
+      settings[feature] &&
+      (!features[feature].test ||
+      features[feature].test(parsedUrl, document.title || ''))
     )
+    document.documentElement.setAttribute('data-refined-twitter-lite', enabledFeatures.join(' '))
   }
 
-  let currentUrl = window.location.href
-  document.body.addEventListener('DOMNodeInserted', () => {
-    if (window.location.href === currentUrl) {
-      return
-    }
-    currentUrl = window.location.href
-    onNavigate(currentUrl)
-  })
-  onNavigate(currentUrl)
-
   document.head.insertAdjacentHTML('beforeend', `
     <style>
-      [data-refined-twitter-lite~="singleColumn"] [data-testid="sidebarColumn"] {
-        display: none;
-      }
-      [data-refined-twitter-lite~="singleColumn"] .rn-1ye8kvj {
-        min-width: 678px;
-      }
+      ${Object.entries(features).map(([feature, data]) =>
+        data.styles.map(rule =>
+          rule.split(',').map(rule =>
+            `[data-refined-twitter-lite~="${feature}"] ${rule.trim()}`
+          ).join(',')
+        ).join('')
+      ).join("\n")}
     </style>
   `)
+
+  document.body.addEventListener('DOMNodeInserted', () => {
+    if (window.location.href !== prevUrl) {
+      setFeatures()
+      prevUrl = window.location.href
+    }
+  })
+  setFeatures()
+
+  // Customize/Save settings API
+
+  window.setRefinedTwitterLiteFeatures = features => {
```

Comment:
I need to figure out why this is not exposed to the global object

Reply:
https://stackoverflow.com/questions/54840147/expose-function-from-userscript?noredirect=1#comment96454107_54840147 I tried to inject an inline script but twitter has CSP protection and blocks that. Will figure out something in the future.